### DISCUSSION TITLE
Emby jail template

### DISCUSSION THREAD

**Discussion by ALeXXBody** [Jip-Hop/jailmaker#173] [Show and tell]
Hi guys. Here is my first template to install and run the latest version of Emby server as a jail on top of Debian without the need of docker.
Is working great and is more responsive than the docker version, at least for me. 
Intel iGPU transcoding is working by default.

```
startup=1
gpu_passthrough_intel=1
gpu_passthrough_nvidia=1
seccomp=1

**# !!!Replace XXX with your network card!!! 
# MODIFY BEFORE STARTTING THE JAIL FOR THE FIRST TIME**
systemd_nspawn_user_args=--network-macvlan=XXX
            --resolv-conf=bind-host
            --bind='/path/to/your/media/and/config:/media'
 
post_stop_hook=

distro=debian
release=bookworm

initial_setup=#!/usr/bin/bash
    set -euo pipefail 
    apt-get update && apt-get -y install wget nano
    mkdir /home/embyserver/ && cd /home/embyserver
    wget https://github.com/MediaBrowser/Emby.Releases/releases/download/4.8.6.0/emby-server-deb_4.8.6.0_amd64.deb
    dpkg -i emby-server-deb_4.8.6.0_amd64.deb
    rm -r /home/embyserver

systemd_run_default_args=--collect
        --property=Delegate=yes
        --property=RestartForceExitStatus=133
        --property=SuccessExitStatus=133
        --property=TasksMax=infinity
        --property=Type=notify
        --setenv=SYSTEMD_NSPAWN_LOCK=0
        --property=KillMode=mixed

systemd_nspawn_default_args=--bind-ro=/sys/module
        --boot
        --inaccessible=/sys/module/apparmor
        --quiet
        --keep-unit
```

Maybe it is useful for someone who wants to do the same as me, not to relay on docker inside of a jail.

**Comment by Jip-Hop** [comment]
Thanks for sharing!

Why do you install nvidia drivers inside the jail? Those should already be available thanks to setting `gpu_passthrough_nvidia=1 `.

Also, do you require this for emby?
```
pre_start_hook=#!/usr/bin/bash
     set -euo pipefail
     echo 'PRE_START_HOOK_EXAMPLE'
     echo 1 > /proc/sys/net/ipv4/ip_forward
     modprobe br_netfilter
     echo 1 > /proc/sys/net/bridge/bridge-nf-call-iptables
     echo 1 > /proc/sys/net/bridge/bridge-nf-call-ip6tables
```

That comes from the docker template and I don't think you need to add it if not using docker inside this jail.

Does it really work with the spaces after the `=` and `:`? I usually write these flags without spaces in those places.
```
--bind= '/path/to/your/media/and/config: /media'
```

**Comment by ALeXXBody** [comment]
```
pre_start_hook=#!/usr/bin/bash
     set -euo pipefail
     echo 'PRE_START_HOOK_EXAMPLE'
     echo 1 > /proc/sys/net/ipv4/ip_forward
     modprobe br_netfilter
     echo 1 > /proc/sys/net/bridge/bridge-nf-call-iptables
     echo 1 > /proc/sys/net/bridge/bridge-nf-call-ip6tables
```
I'm a noob when it comes to Linux and jailmaker so I don't really know what I'm doing, but the desire to learn is  there.
I thought this part helps to create the network connections with the jail. I suppose you don't need it.

> Does it really work with the spaces after the = and :? I usually write these flags without spaces in those places.

yes, here is a print screen from my working jail:
![image](https://github.com/Jip-Hop/jailmaker/assets/93947665/d5445688-ce4e-485b-99ff-286fe311bd2d)

> Why do you install nvidia drivers inside the jail? Those should already be available thanks to setting gpu_passthrough_nvidia=1 .

I will try to create another jail without Nvidia drivers to see how is working

**Reply by Jip-Hop** [reply]
> yes, here is a print screen from my working jail:

Your screenshot looks good, since there are no spaces after the `=` and `:`. I think you should remove the spaces from the template too.

![image](https://github.com/Jip-Hop/jailmaker/assets/2871973/d3d896f5-b055-4030-90dd-f4450229b244)

**Comment by ALeXXBody** [comment]
Updated! Thank you for opening my eyes! Done some cleaning as well.

**Reply by Jip-Hop** [reply]
Looks much better now :) Once you've confirmed you don't need to install the nvidia drivers then you can also remove this instruction:

> To install Nvidia drivers I used this commands inside the jail (jlmkr shell ):

I'm surprised that even worked for you given the fact that these files, such as `nvidia-smi` should be **read only** bind mounted from the host already due to `gpu_passthrough_nvidia=1`. So I'd expect trying to install `nvidia-smi` would throw all kinds of errors.

**Comment by ALeXXBody** [comment]
> `Why do you install nvidia drivers inside the jail? Those should already be available thanks to setting gpu_passthrough_nvidia=1 .`

Yes, you are right, you don't have to install Nvidia drivers. If I remember well, I tried `nvidia-smi` in a newly created jail and was unable to find it.
Thank you for clarifying some things. Now I know more than 2 hours ago. :D
